sources/Afup/Comptabilite/Facture.php

```diff
@@ -599,4 +602,18 @@ function envoyerFacture($reference)
 
         return $ok;
     }
+
+    public function encryptLink($data)
+    {
+        $iv = openssl_random_pseudo_bytes(openssl_cipher_iv_length($this->cipher));
+
+        return base64_encode(openssl_encrypt($data, $this->cipher, $this->key, 0, $iv) . '::' . $iv);
+    }
+
+    public function decryptLink($data)
+    {
+        list($encrypted_data, $iv) = explode('::', base64_decode($data));
```

Comment:
Gestion des erreurs liées à base64_decode et l'absence du séparateur
```suggestion
        $decoded = base64_decode(str_replace(['-', '_'], ['+', '/'],$data));
        if ($decoded === false || strpos($decoded, '::') === false) {
            return false;
        }
        list($encrypted_data, $iv) = explode('::', $decoded);
```